### PR TITLE
Reenable RealmResults remove methods + document behavior

### PR DESCRIPTION
**Pull request by cmelchior**
Reenabled remove methods and documented their behaviour if called while inside loops.

@bmunkholm @kneth @emanuelez

### CODE REVIEW

File: realm/src/main/java/io/realm/RealmResults.java [hunk 1]
```diff
@@ -435,33 +435,36 @@ public double average(String fieldName) {
     // Deleting
 
     /**
-     * Removes an object at a given index.
+     * Removes an object at a given index. This also deletes the object from the underlying Realm.
+     *
+     * Using this method while iterating the list, can result in a undefined behavior. Use
```
Comment (kneth) [inline realm/src/main/java/io/realm/RealmResults.java]: "list, can" -> list can"

**Comment by kneth** [comment]
:+1:

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
+0.79
+ * RealmResults.remove(index) and RealmResults.removeLast() are now working again.
```
Comment (emanuelez) [inline changelog.txt]: I would rephrase it:
 * Re-enabled RealmResults.remove(index) and RealmResults.removeLast()

**Comment by emanuelez** [comment]
:+1:
